packages/fuel-gauge/src/abi/abi-coder.test.ts

```diff
@@ -0,0 +1,1236 @@
+import { Contract, FuelError, Interface } from 'fuels';
+import type {
+  AssetId,
+  BigNumberish,
+  DecodedValue,
+  EvmAddress,
+  RawSlice,
+  WalletUnlocked,
+} from 'fuels';
+import { expectToThrowFuelError, launchTestNode } from 'fuels/test-utils';
+
+import { AbiContract, AbiContractFactory } from '../../test/typegen';
+import { EnumWithNativeInput, ExternalEnumInput } from '../../test/typegen/contracts/AbiContract';
+import type {
+  EnumWithBuiltinTypeInput,
+  EnumWithBuiltinTypeOutput,
+  EnumWithVectorInput,
+  EnumWithVectorOutput,
+  IdentityInput,
+  IdentityOutput,
+  StructDoubleGenericInput,
+  StructSimpleInput,
+  StructSimpleOutput,
+  StructWithGenericArrayInput,
+  StructWithMultiOptionInput,
+  StructWithMultiOptionOutput,
+  StructCInput,
+  StructWithNestedArrayInput,
+  StructWithNestedTupleInput,
+  StructSingleGenericInput,
+} from '../../test/typegen/contracts/AbiContract';
+import type { Option, Result, Vec } from '../../test/typegen/contracts/common';
+
+import {
+  U16_MAX,
+  U16_MIN,
+  U256_MAX,
+  U256_MIN,
+  U32_MAX,
+  U32_MIN,
+  U64_MAX,
+  U64_MIN,
+  U8_MAX,
+  U8_MIN,
+} from './constants';
+import { toEqualBn } from './vitest.matcher';
+
+expect.extend({ toEqualBn });
+
+/**
+ * @group node
+ */
+describe('AbiCoder', () => {
+  let contract: AbiContract;
+  let wallet: WalletUnlocked;
+  let cleanup: () => void;
+
+  beforeAll(async () => {
+    const launched = await launchTestNode({
+      contractsConfigs: [{ factory: AbiContractFactory }],
+    });
+
+    const oldAbi = new Interface(AbiContract.abi);
+
+    wallet = launched.wallets[0];
+    contract = new Contract(launched.contracts[0].id, oldAbi, wallet) as AbiContract;
```

Comment:
```suggestion
    const { contracts: [ contract ] } = launched;
```

And you can remove the `wallet` variable.